resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -246,13 +253,57 @@ private[yarn] class ExecutorRunnable(
       sys.env.get("SPARK_USER").foreach { user =>
         val containerId = ConverterUtils.toString(c.getId)
         val address = c.getNodeHttpAddress
-        val baseUrl = s"$httpScheme$address/node/containerlogs/$containerId/$user"
 
-        env("SPARK_LOG_URL_STDERR") = s"$baseUrl/stderr?start=-4096"
-        env("SPARK_LOG_URL_STDOUT") = s"$baseUrl/stdout?start=-4096"
+        val customLogUrl = sparkConf.get(config.CUSTOM_LOG_URL)
+
+        val envNameToFileNameMap = Map("SPARK_LOG_URL_STDERR" -> "stderr",
+          "SPARK_LOG_URL_STDOUT" -> "stdout")
+        val logUrls = ExecutorRunnable.buildLogUrls(customLogUrl, httpScheme, address,
+          clusterId, containerId, user, envNameToFileNameMap)
+        logUrls.foreach { case (envName, url) =>
+          env(envName) = url
+        }
       }
     }
 
     env
   }
 }
+
+private[yarn] object ExecutorRunnable {
+  def buildLogUrls(
+    logUrlPattern: String,
+    httpScheme: String,
+    nodeHttpAddress: String,
+    clusterId: Option[String],
+    containerId: String,
+    user: String,
+    envNameToFileNameMap: Map[String, String]): Map[String, String] = {
```

Comment:
4-space indent for method parameters

Reply:
Ah yes I just confused indent rule between method parameters and return... Nice catch. Will address.